test/format_unittest.cc

```diff
@@ -0,0 +1,42 @@
+#include "../src/format.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(SubstituteTest, ValidFormat) {
+  const std::string endpoint_format =
+      "https://stackdriver.googleapis.com/v1beta2/projects/{{project_id}}";
```

Comment:
I wonder if we need to keep the original format string here. How about something like:
```cpp
  const std::string format_str("prefix|{{subst}}|postfix");
  EXPECT_EQ("prefix|value|postfix",
            format::Substitute(format_str, {{"subst", "value"}}));
```
? Ditto for other tests.

Reply:
Doesn't have to be. But it makes the test more descriptive of the use cases.
Updated.